storage/replica_command.go

```diff
@@ -546,12 +550,29 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
 		// less than "f", and so the meta row {"f"->["c", "f")} would be
 		// ignored by MVCCReverseScan. In the second case, the range descriptor
 		// received by MVCCScan will be filtered before results are returned.
+		// Example: ranges ["c", "f") and ["f", "z"), reverse scan on ["d", "f"]
+		// leads to inspecting the descriptor {"z"->["f","z")}, which is
+		// discarded since it's not being asked for.
 		startKey, endKey := keys.MetaScanBounds(args.Key)
 		kvs, intents, err = engine.MVCCScan(batch, startKey, endKey, 1,
 			args.Timestamp, consistent, args.Txn)
 		if err != nil {
 			return reply, nil, err
 		}
+		var firstRD proto.RangeDescriptor
+		if err = gogoproto.Unmarshal(kvs[0].Value.Bytes, &firstRD); err != nil {
+			return reply, nil, err
+		}
+		if keys.RangeMetaKey(firstRD.StartKey).Less(args.Key) {
```

Comment:
Sorry, got confused... When `args.Key` is `"meta2b"`, we will first scan `["meta2b".Next(), "meta2\xff\xff")` then will reverse scan `["meta2", "meta2b".Next())`. The first scan hits range `["b", "c")`, and the second scan hits range `["a", b")`. Should we return both?